enumerable.rb

```diff
@@ -0,0 +1,134 @@
+# frozen_string_literal: true
+
+module Enumerable
+  def my_each
+    if block_given?
+      (0..length - 1).my_each do |i|
+        yield(self[i])
+      end
+    else
+      puts "You didn't pass in a block!"
+    end
+  end
+
+  def my_each_with_index
+    if block_given?
+      (0..length - 1).my_each do |i|
+        yield(self[i], i)
+      end
+    else
+      puts "You didn't pass in a block!"
+    end
+  end
+
+  def my_select
+    if block_given?
+      new_arr = []
+      (0..length - 1).my_each do |i|
+        new_arr.push(self[i]) if yield(self[i])
+      end
+    else
+      puts "You didn't pass in a block!"
+    end
+  end
```

Comment:
- We need to fix my_each first to make this method works.

- My_select should returns an array containing all elements of enum for which the given block returns a true value.
`[1,4,5].select { |x| x > 2}`
`=> [4, 5]`

- My_select returns an enumerator if no block is given.
`[1,4,5].select `
`=> #<Enumerator: [1, 4, 5]:select>`